console/v1/types_console_quick_start.go

```diff
@@ -0,0 +1,126 @@
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:MinItems=1
+	// +required
+	Tasks []ConsoleQuickStartTask `json:"tasks"`
+	// conclusion sums up the Quick Start and suggests the possible
+	// next Quick Starts for the user to try. (includes markdown)
+	// +optional
+	Conclusion string `json:"conclusion,omitempty"`
+	// nextQuickStart is a list of the following Quick Starts.
+	// +optional
+	NextQuickStart []string `json:"nextQuickStart,omitempty"`
+	// accessReviewResources containes arrays of resources that user's access
+	// will be reviewed against in order for the user to complete the Quick Start.
+	// The Quick Start will be hidden if any of the access reviews fail.
+	// +optional
+	AccessReviewResources []authorizationv1.ResourceAttributes `json:"accessReviewResources,omitempty"`
+}
+
+// ConsoleQuickStartTask is a single step in a Quick Start.
+type ConsoleQuickStartTask struct {
+	// title describes the task and is displayed as a step heading.
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:MinLength=1
+	// +required
+	Title string `json:"title"`
+	// description describes the steps needed to complete the task. (includes markdown)
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:MinLength=1
+	// +required
+	Description string `json:"description"`
+	// review contains instructions to validate the task is complete. The user will select 'Yes' or 'No'
+	// using a radio button, which indicates whether the step was completed successfully.
+	// +optional
+	Review *ConsoleQuickStartTaskReview `json:"review,omitempty"`
+	// recapitulation contains information about passed step.
+	// +optional
+	Recapitulation *ConsoleQuickStartTaskRecapitulation `json:"recapitulation,omitempty"`
+}
+
+// ConsoleQuickStartTaskReview contains instructions that validate a task was completed successfully.
+type ConsoleQuickStartTaskReview struct {
+	// instructions contains steps that user needs to take in order
+	// to validate his work after going through a task. (includes markdown)
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:MinLength=1
+	// +required
+	Instructions string `json:"instructions"`
+	// failedTaskHelp contains suggestions for a failed task review and is shown at the end of task.
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:MinLength=1
+	// +required
+	FailedTaskHelp string `json:"failedTaskHelp"`
+}
+
+// ConsoleQuickStartTaskRecapitulation contains information about a passed step.
+type ConsoleQuickStartTaskRecapitulation struct {
+	// success describes the succesfully passed task.
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:MinLength=1
+	// +required
+	Success string `json:"success"`
+	// failed describes the unsuccessfully passed task.
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:MinLength=1
+	// +required
+	Failed string `json:"failed"`
```

Comment:
We should indicate that these should be short messages.
Perhaps:
```
// failed briefly describes the unsuccessfully passed task.
```

We also have a need to render links, therefore markdown is supported.